### ISSUE TITLE
Improve subtitle merge algorithm

### ISSUE THREAD

**Issue by oliverfei** [oliverfei/videocr-PaddleOCR#5]
The algorithm currently merges 2 subtitles by picking the one with the highest average confidence.
Ideally, all characters with high confidence should be used if there is no conflict in their positions e.g.

If merging `a (99) b (99) c (99)` and `a (99) b (99) a (98) c (99)`, the result should be `abac` instead of `abc`

**Comment by obedabit** [comment]
Is this the reason why the subtitle appears wrong ( flipped and sometimes in the wrong order)?

**Comment by oliverfei** [comment]
@obedabit do you have an example?

**Comment by obedabit** [comment]
@oliverfei Yes I do, here is an example, I have the following photo it has two line sentences, and the output is wrong on one line, it should be two lines and the last sentence should be the first.

![image](https://user-images.githubusercontent.com/54214435/203988736-1ac490c7-5905-49d4-8a6e-a4e3452ec830.png)

output:

3
00:00:51,416 --> 00:00:55,583
right in front of you, of course you'd be depressed. No, if someone else kissed the person you liked

**Comment by oliverfei** [comment]
Ah ok. That's cause the library doesn't account the multiline scenario at the moment which is unrelated to this issue.

**Comment by obedabit** [comment]
so should we make an issue for It or you will fix it without one

**Comment by oliverfei** [comment]
I pushed something if you want to try it out

**Comment by obedabit** [comment]
It is ok, amazing work

**Comment by woojh3690** [comment]
The subtitle merging algorithm used by this library is too bad for subtitles that appear as if they were typed rather than appearing in a single frame.
